src/plugins/explore/public/components/visualizations/heatmap/heatmap_vis_config.ts

```diff
@@ -33,7 +35,7 @@ export interface ExclusiveHeatmapConfig {
   scaleToDataBounds: boolean;
   percentageMode: boolean;
   maxNumberOfColors: number;
-  useCustomRanges: boolean;
```

Comment:
Instead of deleting `useCustomRanges`,  make it optional and mark it as `@deprecated`.